doc/telestrat.sgml

```diff
@@ -0,0 +1,177 @@
+<!doctype linuxdoc system>
+
+<article>
+
+<title>Oric Telestrat-specific information for cc65
+<author>
+<url url="mailto:jede@oric.org" name="Jede">,<newline>
+
+<date>2017-01-22
+
+<abstract>
+An overview over the Telestrat (telemon 3.0 : http://orix.oric.org) runtime system as it is implemented for the cc65 C
+compiler.
+</abstract>
+
+<!-- Table of contents -->
+<toc>
+
+<!-- Begin the document -->
+
+<sect>Overview<p>
+
+This file contains an overview of the Telestrat runtime system as it comes with the
+cc65 C compiler. It describes the memory layout, Telemon3.0-specific header files,
+available drivers, and any pitfalls specific to that platform.
+
+Please note that Telemon3.0-specific functions are just mentioned here, they are
+described in detail in the separate <url url="funcref.html" name="function
+reference">. Even functions marked as "platform dependent" may be available on
+more than one platform. Please see the function reference for more
+information.
+
+
+
+<sect>Binary format<p>
+
+The standard binary output format generated by the linker for the Telemon 3.0 target
+is a machine language program with a 20 bytes header described here : http://orix.oric.org/doku.php?id=orix:header
```

Comment:
- You can't presume the Telemon 3.0 thing to be known. You need to elaborate much more about it. The link given doesn't explain _ANYTHING_. I guess that the Telemon 3.0 uses the .o65 file format. If yes: Why? Is the file to be placed at different locations? Are references into ROM/OS/DOS routines not at fixed locations? I think you get the point of info I'd like to either see here or on a page @ orix.oric.org and linked to from here.

- If this isn't only about Telemon 3.0 then - as far as I understand - you need (at least) two different binary formats. One for the Telemon 2.4 (what ever that might look like) and the .o65 (?) one. Both need to be described here. And you need to find a way to have both generated. Usually this is done via a) parameters in the .cfg file or b) several .cfg files. Check out the Apple II and ATARI sections @cc65 to get an idea on multiple .cfg files for different binary formats. If it's only about the header then you might get away with one .cfg file and a command line parameter. See http://cc65.github.io/doc/apple2.html#ss4.4 for an example of an optional header.

- If there's a binary format compatible with both 2.4 and 3.0 that one should be the default. The 3.0 specific binary format needs to be the "additional option" - even if that file format (and the whole 3.0 thing in general) should be your personal motivation to add a target to cc65 in the first place.